source/Engine.cpp

```diff
@@ -1625,8 +1625,22 @@ void Engine::HandleKeyboardInputs()
 	else
 		activeCommands.Clear(Command::WAIT);
 	
-	// Transfer all newly pressed, unhandled keys to active commands.
-	activeCommands |= keyDown;
+	// Translate shift+BACK to a command to a STOP command to stop all movement of the flagship.
+	// Translation is done here to allow the autoPilot (which will execute the STOP-command) to
+	// act on a single STOP command instead of the BACK+SHIFT modifier).
+	if(keyHeld.Has(Command::BACK) && keyHeld.Has(Command::SHIFT))
+		activeCommands |= Command::STOP;
+	else if(keyHeld.Has(Command::BACK))
+		activeCommands |= Command::BACK;
+	
+	// Transfer all commands that need to be active as long as the corresponding key is pressed.
+	activeCommands |= (keyHeld.And(Command::PRIMARY | Command::SECONDARY | Command::SCAN |
```

Comment:
I think this line is partially duplicating what is done in lines 1601-1603

Reply:
Agreed. I merged the two statements and did a short re-test. The movement keys and stop command still appear to be working fine after merging those.